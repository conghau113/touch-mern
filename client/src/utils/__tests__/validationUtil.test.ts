import { toMatchImageSnapshot } from 'jest-image-snapshot'
import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest'

declare global {
  namespace jest {
    interface Matchers<R> {
      toMatchImageSnapshot(): R
    }
  }
}

expect.extend({ toMatchImageSnapshot })

const businessHours = [9, 17]

const purchase = () => {
  const currentHour = new Date().getHours()
  const [open, close] = businessHours

  if (currentHour > open && currentHour < close) return { message: 'Success' }

  return { message: 'Error' }
}

describe('purchasing flow', () => {
  beforeEach(() => {
    // tell vitest we use mocked time
    vi.useFakeTimers()
  })

  afterEach(() => {
    // restoring date after each test run
    vi.useRealTimers()
  })

  it('allows purchases within business hours', () => {
    // set hour within business hours
    const date = new Date(2000, 1, 1, 13)
    vi.setSystemTime(date)

    // access Date.now() will result in the date set above
    expect(purchase()).toEqual({ message: 'Success' })
  })

  it('allows purchases within business hours', () => {
    // set hour within business hours
    const date = new Date(2000, 1, 1, 13)
    vi.setSystemTime(date)

    // access Date.now() will result in the date set above
    expect(purchase()).toEqual({ message: 'Success' })
  })

  it('allows purchases within business hours', () => {
    // set hour within business hours
    const date = new Date(2000, 1, 1, 13)
    vi.setSystemTime(date)

    // access Date.now() will result in the date set above
    expect(purchase()).toEqual({ message: 'Success' })
  })

  it('disallows purchases outside of business hours', () => {
    // set hour outside business hours
    const date = new Date(2000, 1, 1, 19)
    vi.setSystemTime(date)

    // access Date.now() will result in the date set above
    expect(purchase()).toEqual({ message: 'Error' })
  })

  it('test equal', () => {
    expect(1 + 3).toBe(4)
  })

  it('toUpperCase', () => {
    const result = 'foobar'.toUpperCase()
    expect(result).toMatchSnapshot()
  })

  it('toUpperCase testing', () => {
    const result = 'testing ne'.toUpperCase()
    expect(result).toMatchSnapshot()
  })
})
