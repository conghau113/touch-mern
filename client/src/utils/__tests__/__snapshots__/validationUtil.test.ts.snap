// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`purchasing flow > toUpperCase 1`] = `"FOOBAR"`;

exports[`purchasing flow > toUpperCase 1 1`] = `"\\"FOOBAR\\""`;

exports[`purchasing flow > toUpperCase testing 1`] = `"TESTING NE"`;
